sihl_contrib/session/lib/middleware.ml

```diff
@@ -1,11 +1,51 @@
-(* TODO
-https://docs.djangoproject.com/en/3.0/topics/http/sessions/
-*)
+let ( let* ) = Lwt.bind
 
-let m app =
+let cookie_key = "sessions_session_id"
+
+let session () app =
   let filter handler req =
-    Logs.warn (fun m -> m "session middleware is not implemented");
-    handler req
+    let (module Repository : Repo_sig.REPOSITORY) =
+      Sihl.Core.Registry.get Bind.Repository.key
+    in
```

Comment:
I'm still not sure we should have a generic repository concept. All repositories and their queries are application specific and should be used only be application specific service(s). So here the repository, should be SessionRepository.

I think we need to review this approach and discuss more about it.